airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_stream_reader.py

```diff
@@ -54,11 +54,22 @@ def filter_files_by_globs(files: List[RemoteFile], globs: List[str]) -> List[Rem
         """
         Utility method for filtering files based on globs.
         """
-        return [file for file in files if any(fnmatch(file.uri, glob) for glob in globs)]
+        if not globs:
+            return files
+
+        matches = {}
+
+        for file in files:
+            for g in globs:
+                if globmatch(file.uri, g, flags=GLOBSTAR):
+                    matches[file.uri] = file
+
+        return list(matches.values())
 
     @staticmethod
-    def get_prefixes_from_globs(files: List[RemoteFile], globs: List[str]) -> List[str]:
+    def get_prefixes_from_globs(globs: List[str]) -> List[str]:
```

Comment:
can we add unit tests for this method?

Reply:
They're rolled into the unit tests for `filter_files_by_globs`. Updated the test name to make that clearer.